src/trainer.py

```diff
@@ -509,26 +511,14 @@ def train(
 
         # define these here b/c they might get overridden on load
         n_pass, should_stop = 0, False
-        if self._serialization_dir is not None and phase == "pretrain":
+        if self._serialization_dir is not None:
             # Resume from serialization path
-            if load_model and any(
-                ["model_state_" in x for x in os.listdir(self._serialization_dir)]
-            ):
-                n_pass, should_stop = self._restore_checkpoint()
+            if load_model:
+                n_pass, should_stop = self._restore_checkpoint(phase, tasks)
```

Comment:
We should check that there's a checkpoint before loading: load_model is on by default for good reason, and this crashes if load_model is set but there's no checkpoint. (I had to manually switch this to get a test to run.)

Reply:
Alright, I'll just put a log warning here if load_model=1 but there's no checkpoint. 
That problem should only be hit if the user put in load_model=1 though when there is no checkpoint.

Reply:
There shouldn't be a warning at all. What I'm trying to say is that the most normal/typical/ideal experiment should start with load_model = 1 and no checkpoint. It's only off in defaults.conf since that makes it easier to catch bugs.